pkgs/top-level/all-packages.nix

```diff
@@ -1147,7 +1147,10 @@ with pkgs;
   makeInitrdNG = callPackage ../build-support/kernel/make-initrd-ng.nix;
   makeInitrdNGTool = callPackage ../build-support/kernel/make-initrd-ng-tool.nix { };
 
-  makeWrapper = makeShellWrapper;
+  makeWrapper =
+    if stdenv.hostPlatform.isWindows || stdenv.hostPlatform.isEmscripten
+    then null
```

Comment:
should throw or something

comment why it doesn't work on emscripten

can't be changed to makeBinaryWrapper on windows https://github.com/NixOS/nixpkgs/issues/120726#issuecomment-1518984250

Reply:
My intent here is to allow packages to conditionally use `makeWrapper` if it is available by checking for `makeWrapper == null`. Changing this to a throw wouldn't work for that. Is there a better way to do that? Maybe something like `lib.elem stdenv.hostPlatform makeWrapper.meta.platforms`?

Reply:
Indeed, the using expression should check whether something is available or not. `lib.meta.availableOn` should suffice for this, given that the compatibility is expressed in (`makeWrapper`'s) `meta.platforms`.

Using `null` will also break `makeWrapper` in `buildPackages` for the affected package sets due to splicing depending on attribute sets.